rviz_common/include/rviz_common/ros_integration/ros_node_abstraction_iface.hpp

```diff
@@ -45,10 +54,20 @@ namespace ros_integration
 class RosNodeAbstractionIface
 {
 public:
-  virtual std::string get_node_name() = 0;
+  using WeakPtr = std::weak_ptr<RosNodeAbstractionIface>;
+
+  virtual ~RosNodeAbstractionIface() = default;
+
+  virtual void add_to_executor(rclcpp::executor::Executor & executor) = 0;
+
+  virtual std::string get_node_name() const = 0;
 
   virtual std::map<std::string, std::vector<std::string>>
-  get_topic_names_and_types() = 0;
+  get_topic_names_and_types() const = 0;
+
+  // TODO(anhosi): remove once the RosNodeAbstraction is extended to handle subscriptions
```

Comment:
Maybe this is redundant information with what I've written above, but this todo isn't sufficient. You'd also need to remove the reference to the executor above.